set.mm

```diff
@@ -505773,6 +506254,424 @@ codes is an increasing chain (with respect to inclusion).  (Contributed
       HUOGZVHVGNUTURWBUSCHUNULCUPRUQ $.
   $}
 
+  $( Lemma for ~ satffunlem1lem1 and ~ satffunlem2lem1 .  (Contributed by AV,
+     27-Oct-2023.) $)
+  satffunlem $p |- ( ( ( Fun Z /\ ( s e. Z /\ r e. Z )
+                                /\ ( u e. Z /\ v e. Z ) )
+             /\ ( x = ( ( 1st ` s ) |g ( 1st ` r ) )
+                  /\ y = ( ( M ^m _om ) \ ( ( 2nd ` s ) i^i ( 2nd ` r ) ) ) )
+             /\ ( x = ( ( 1st ` u ) |g ( 1st ` v ) )
+                  /\ w = ( ( M ^m _om ) \ ( ( 2nd ` u ) i^i ( 2nd ` v ) ) ) ) )
+                      -> y = w ) $=
+    ( cv wcel wa c1st cfv wceq c2nd wi c1o cop cvv wfun w3a cgna co com cin weq
+    cmap cdif eqtr2 fvex gonafv mp2an eqeq12i 1oex opex opth anbi2i funfv1st2nd
+    3bitri ex anim12d fveq2 eqcoms adantr eqeq1d adantl anbi12d anbi1d ad2ant2r
+    ad2ant2l ineq12d syl6bi com12 a1i syl2and 3imp1 difeq2d eqeq12 mpbird exp43
+    expd wb adantld syl5bi syl5 com35 impd com24 3imp ) GUAZHJZGKZIJZGKZLZEJZGK
+    ZDJZGKZLZUBZAJZWLMNZWNMNZUCUDZOZBJZFUEUHUDZWLPNZWNPNZUFZUIZOZLXCWQMNZWSMNZU
+    CUDZOZCJZXIWQPNZWSPNZUFZUIZOZLZBCUGZXBXGXNYEYFQXBYEXNXGYFXBXRYDXNXGYFQQXBXR
+    XGXNYDYFXBXRXGXNYDYFQQZXRXGLXQXFOZXBYGXCXQXFUJYHRROZXOXDOZXPXEOZLZLZXBYGYHR
+    XOXPSZSZRXDXESZSZOYIYNYPOZLYMXQYOXFYQXOTKXPTKXQYOOWQMUKZWSMUKZXOXPTTULUMXDT
+    KXETKXFYQOWLMUKWNMUKXDXETTULUMUNRYNRYPUOXOXPUPUQYRYLYIXOXPXDXEYSYTUQURUTXBY
+    LYGYIXBYLXNYDYFXBYLLZXNYDLZLYFXMYCOZUUAUUCUUBUUAXLYBXIWKWPXAYLXLYBOZWKWPXAY
+    LUUDQZWKWPXDGNZXJOZXEGNZXKOZLZXAXOGNZXTOZXPGNZYAOZLZUUEWKWMUUGWOUUIWKWMUUGG
+    WLUSVAWKWOUUIGWNUSVAVBWKWRUULWTUUNWKWRUULGWQUSVAWKWTUUNGWSUSVAVBUUJUUOLZUUE
+    QWKYLUUPUUDYLUUPUUKXJOZUUMXKOZLZUUOLZUUDYLUUJUUSUUOYLUUGUUQUUIUURYLUUFUUKXJ
+    YJUUFUUKOZYKUVAXDXOXDXOGVCVDVEVFYLUUHUUMXKYKUUHUUMOZYJUVBXEXPXEXPGVCVDVGVFV
+    HVIUUTXJXTXKYAUUQUULXJXTOUURUUNUUKXJXTUJVJUURUUNXKYAOUUQUULUUMXKYAUJVKVLVMV
+    NVOVPWBVQVRVEUUBYFUUCWCUUAXHXMXSYCVSVGVTWAWDWEWFWBWGWHWIWHWJ $.
+
+  ${
+    $d E i j s u x y z $.  $d E i r s u x y z $.  $d E j s u v x y z $.
+    $d M i j s u x y z $.  $d M i r s u x y z $.  $d M j s u v x y z $.
+    $d N i j s u x y z $.  $d N i r s u x y z $.  $d N j s u v x y z $.
+    $d f i j s u y z $.  $d f i r s u y z $.  $d f j s u v y z $.
+    $d i j k s u y z $.  $d k r s u v y z $.
+    $( Lemma for ~ satffunlem1 .  (Contributed by AV, 17-Oct-2023.) $)
+    satffunlem1lem1 $p |- ( Fun ( ( M Sat E ) ` N )
```

Comment:
I understand what you're doing with `lem1lem1`, `lem2lem1`, usw. 
I think usually we are simply numbering `lem1` through `lem6` instead. 
Anyway I'm guilty of too (I've used `lema`, etc.)

Reply:
I wouldn't especially push the numbers (they are kind of awkward especially if lemmas get added or removed) but I would advocate a name which avoids multiple levels of `lem`. Maybe `satfunlem1a` or `satfunlemsat` or something.

Reply:
Reading the conventions about lemmas:

>        A proof with the sole purpose of supporting a final proof is a lemma;
>        the naming convention for a lemma is the final proof label followed by
>        "lem", and a number if there is more than one.  E.g., ~ sbthlem1 is the
>        first lemma for ~ sbth .  The comment should begin with "Lemma for",
>        followed by the final proof label, so that it can be suppressed in
>        theorem lists (see the Metamath program "MM> WRITE THEOREM_LIST"
>        command).

my names obey these rules: the final proof (theorem) label is ~satffunlem1, and the sole purpose of ~satffunlem1lem1 is to support ~satffunlem1. 

Having a lot of lemmas, I think it is helpful for the understanding to allow at least one level of nesting (lemmas of lemmas). I agree that a deeper nesting (lemmas of lemmas of lemmas, etc.) would be more confusing than helpful.

Reply:
> Having a lot of lemmas, I think it is helpful for the understanding to allow at least one level of nesting (lemmas of lemmas). I

In that case I think we should go with your preference on that. Especially once we hit "lem" I think we can be flexible on these names.